src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -827,5 +833,42 @@ public void Dispose()
                 disposed = true;
             }
         }
+
+        internal async Task HandleCertificateMigrationRequest(
```

Comment:
I would suggest wrapping the whole request handler in a try/catch then use `requestContext.SendError()` to send back any exceptions, sort of like this:

https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs#L302